### DISCUSSION TITLE
Ardupilot on 64 bit Pi Zero 2, Jetson Nano/Orin, and OBAL

### DISCUSSION THREAD

**Discussion by b06stiles** [HefnySco/OBAL#47] [Ideas]
Hi All

Firstly, thanks to Mohammad for his fantastic work!

So I've looking at getting Ardupilot running on the nVidia Jetson series of modules, ie high powered 64-bit linux MCUs.  To do this - as a stepping stone - I followed Mohammad's excellent article on Medium here: 

[https://medium.com/@mohammad.hefny/running-ardupilot-on-raspberry-os-64-bullseye-832f7d28bc22](url)

to compile a custom linux kernel for Raspberry Pi.  This enables access to /dev/mem which is required for access to the GPIO ports.

Here's the rig.  The Nano's in the background with some short ribbon jumper cables to attach to OBALs Pi header.  The Nano's GPIO's can't drive long cables - they tend to oscillate.

![IMG_20231107_095544516](https://github.com/HefnySco/OBAL/assets/13236678/663a23a6-24c2-4072-8002-7ce331446d07)

Up to now I've got both 64 bit Bullseye and 64 bit Ubuntu working with the Raspberry Pi Zero 2/OBAL, and I have Ardupilot running on a Jetson Nano, however there remain unresolved problems with the SPI driver on that device.

To anyone who might want to follow the instructions in the article, I believe there's a step missing when transferring the newly programmed SD card back into the target Raspberry Pi.  The instructions should read:

```
sudo depmod
sudo modprobe configs
zgrep STRICT_DEVMEM /proc/config.gz
```
otherwise `configs `doesn't exist.

Has anyone had any experience with the Jetson Nano?  The SPI driver seems quite broken, and and Nano is obsolescent, so I'm looking to getting my hands on an Orin NX in the near future.

Thanks and regards
Bob

**Comment by HefnySco** [comment]
Sorry I want to understand something... you are trying to run OBAL on Jetson Nano instead of RPI ?

**Comment by b06stiles** [comment]
Hi Muhammad

Good to hear from you!  Yes, I've done quite a bit of work on it.
Unfortunately I hit a problem with the NANO's SPI driver, which seems to be
broken.
Ultimately I want to try the same thing on an NVIDIA Orin NX, but I don't
have my hands on one yet.

**Reply by HefnySco** [reply]
Well I would think that you may use external AHRS on STM32 or something then serialize data on serial port.
https://medium.com/@mohammad.hefny/ardupilot-linux-based-board-stm32-as-a-breakout-module-5b57989863ce

**Comment by b06stiles** [comment]
Yeah, thanks.  I already saw this.  I guess the plan is to try to eliminate the STM32 side of things (flight controller etc) completely.
I'll update the thread when the ORIN arrives (if it works haha)

The OBAL board has been really useful as a breakout, so thanks for that.

**Reply by HefnySco** [reply]
yes please do.
Thanks
